### ISSUE TITLE
Add explicit links to static pages

### ISSUE THREAD

**Issue by ilg-ul** [doxygen/doxygen#10447]
**Describe the bug**

Static pages do not have explicit links in the home page and validators and other crawlers cannot reach them.

**Expected behavior**

All pages to be reachable by crawlers.

**Screenshots**

As seen in the image, there are several static pages, created with `@page`, which are rendered pretty well:

![Screenshot 2023-11-26 at 23 43 50](https://github.com/doxygen/doxygen/assets/362732/ac2f9865-dbaa-4e04-a10f-c5eda92e96a7)

Three of them have explicit links in the middle of the home page; there are crawled. The others are not.

**To Reproduce**

The site where this problem was detected is <https://micro-os-plus.github.io/micro-test-plus-xpack/>.

**Version**

1.9.8

**Stack trace**

-

**Additional context**

My workaround was to manually add dummy links in the home page, but this is not ok.

![Screenshot 2023-11-26 at 23 44 34](https://github.com/doxygen/doxygen/assets/362732/26c71304-41bf-4438-8b4a-2f6671550a85)

I think that Doxygen should guarantee that these links are always automatically generated, regardless of the user configuration.

**Comment by albert-github** [comment]
The question here is: 
- is this a task for doxygen or is the used crawler / validator not intelligent enough to find these links?

There will be references to the different pages (e.g through the treeview or through links in the different pages) and recursive crawling should be able to find these pages.

I don't think this is a doxygen bug.

**Comment by ilg-ul** [comment]
> or is the used crawler / validator not intelligent enough to find these links?

Ah, this is always a possibility, but I tried several such validators, including <https://validator.w3.org/checklink>, but at no avail. (to be sure, I temporarily disabled my links and retried; there are no references to pages.html, credits.html, etc).

> There will be references to the different pages (e.g through the treeview or through links in the different pages) and recursive crawling should be able to find these pages.

I thought so too, but when I tried to identify these links, they were not there. :-(

For example, in my project I have a `credits.html` page. If I try to search references to it, I get:

```console
ilg@wksi html % grep -r 'credits.html' *        
credits.html:$(document).ready(function(){initNavTree('credits.html',''); initResizable(); });
index.html:<a href="credits.html"/>    <---
navtreedata.js:    [ "Credits", "credits.html", null ],
navtreeindex1.js:"credits.html":[6],
pages.html:<tr id="row_5_" class="odd"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><a class="el" href="credits.html" target="_self">Credits</a></td><td class="desc"></td></tr>
search/all_7.js:  ['credits_28',['Credits',['../credits.html',1,'']]],
search/pages_1.js:  ['credits_1',['Credits',['../credits.html',1,'']]]
```

The only reference in the home page in the dummy link that I manually added.

There is a reference in the table in `pages.html`, but this also is not referred directly from the home page:

```console
ilg@wksi html % grep -r 'pages.html' *  
index.html:<a href="pages.html"/>    <---
menudata.js:{text:"Related Pages",url:"pages.html"},
navtreeindex2.js:"pages.html":[],
pages.html:$(document).ready(function(){initNavTree('pages.html',''); initResizable(); });
```

Similarly, the only reference is the dummy link that I manually added.

All other references are in JavaScript files.

I don't think that it is reasonable to expect validators and crawlers to run the JavaScript code.

Thus, explicitly adding such links to the static pages is a quick workaround.

**Comment by albert-github** [comment]
In your comments (https://github.com/doxygen/doxygen/issues/10447#issuecomment-1827769599) you mention credits.html, is this a doxygen generated files or ..., so:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by albert-github** [comment]
To summarize the example:
- the relevant page is `pages.html`
- the page `pages.html` can only be reached through the "Related Pages" item in the "blue bar": ![image](https://github.com/doxygen/doxygen/assets/5223533/f772e158-7498-4152-8b95-3ba3dbed7be7)

**Comment by ilg-ul** [comment]
> the relevant page is pages.html

The relevant pages are both `pages.html` and `xyz.html`.

> the page pages.html can only be reached through the "Related Pages" item in the "blue bar"

`pages.html` can be reached _by a human_ from the blue bar.

A dumb machine cannot identify a reference to `pages.html` by parsing the home page. Neither to `xyz.html`. Possibly other.

So, as long as access to all custom pages, to `pages.html` and possibly other, is via content dynamically generated by JavaScripts, these pages need artificial/dummy links in the home page for validators/crawlers to reach them.

**Comment by albert-github** [comment]
How would these artificial links in the main page (index.html) look like without disturbing the page?

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by ilg-ul** [comment]
I confirm that with 1.10.0 it seems ok on my projects.
